api/file_test.go

```diff
@@ -820,11 +820,21 @@ func readTestFile(name string) ([]byte, error) {
 	}
 }
 
-func s3New(endpoint, accessKey, secretKey string, secure bool, signV2 bool) (*s3.Client, error) {
+// Similar to s3.New() but allows initialization of signature v2 or signature v4 client.
+// If signV2 input is false, function always returns signature v4.
+//
+// Additionally this function also takes a user defined region, if set
+// disables automatic region lookup.
+func s3New(endpoint, accessKey, secretKey string, secure bool, signV2 bool, region string) (client *s3.Client, err error) {
 	if signV2 {
-		return s3.NewV2(endpoint, accessKey, secretKey, secure)
+		client, err = s3.NewV2(endpoint, accessKey, secretKey, secure)
+	} else {
+		// Region can only be configured if using signature v4, use
+		// mattermost/platform-v4.1. For v2 signature regions are
+		// not quite meaningful and should work fine without.
+		client, err = s3.NewWithRegion(endpoint, accessKey, secretKey, secure, region)
 	}
-	return s3.NewV4(endpoint, accessKey, secretKey, secure)
+	return
```

Comment:
Can you switch this to just return the results of `s3.NewV2` and `s3.NewWithRegion`? I don't think we use named return values anywhere else, and I'd prefer to be consistent